include/fiction/algorithms/physical_design/apply_gate_library.hpp

```diff
@@ -297,7 +314,48 @@ template <typename CellLyt, typename GateLibrary, typename GateLyt, typename Par
 
     detail::apply_gate_library_impl<CellLyt, GateLibrary, GateLyt> p{lyt};
 
-    return p.template run_parameterized_gate_library<Params>(params);
+    // Run the gate library with the parameters
+    const CellLyt result = p.template run_parameterized_gate_library<Params>(params);
+
+    return result;
+}
+
+/**
+ * Applies a defect-aware parameterized gate library to a given
+ * gate-level layout and, thereby, creates and returns a cell-level layout.
+ *
+ * May pass through, and thereby throw, an `unsupported_gate_type_exception`, an
+ * `unsupported_gate_orientation_exception` and any further custom exceptions of the gate libraries.
+ *
+ * @tparam DefectLyt Type of the returned cell-level layout.
+ * @tparam GateLibrary Type of the gate library to apply.
+ * @tparam GateLyt Type of the gate-level layout to apply the library to.
+ * @tparam Params Type of the parameter used for SiDB on-the-fly gate library.
+ * @param lyt The gate-level layout.
+ * @param params Parameter for the gate library.
+ * @param defect_lyt Defect surface.
+ * @return A cell-level layout that implements `lyt`'s gate types with building blocks defined in `GateLibrary`.
+ */
+template <typename DefectLyt, typename GateLibrary, typename GateLyt, typename Params>
+[[nodiscard]] DefectLyt apply_parameterized_gate_library_on_defective_surface(const GateLyt& lyt, const Params& params,
+                                                                              const DefectLyt& defect_surface)
```

Comment:
Question regarding grammar here: "apply" usually goes with the preposition "to". Example: "I **apply** the gate library **to** the layout". I don't know if "on" makes sense here. Could you double-check this, please?

Reply:
Since the library is not directly applied to the surface but the layout, though, maybe "with" could make sense here.

Reply:
I like "to". Many thanks!